File: RootTools2/src/main/java/com/rarnu/tools/neo/api/NativeAPI.java
package com.rarnu.tools.neo.api;

import android.content.Context;
import android.content.Intent;
import com.rarnu.tools.neo.fragment.CleanFragment;

public class NativeAPI {
    static {
        System.loadLibrary("cmd");
    }

    public static boolean isRejected = false;
    public static boolean isSystemRW = false;

    public static native boolean mount();
    public static native boolean isSystemRW();
    public static native void makePreferenceReadable(int sdk, String packageName);
    public static native boolean freezeApplication(String packageName, boolean isFreezed);
    public static native boolean freezeComponent(String packageName, String componentName, boolean isFreezed);
    public static native boolean freezeComponents(String packageName, String[] componentNames, boolean isFreezed);
    public static native void systemClean(Context ctx);
    public static native boolean writeFile(Context ctx, String filePath, String text, int perm);
    public static native boolean catFile(String src, String dest, int perm);
    public static native void forceDeleteFile(String path);
    public static native void forceDropCache();
    public static native void killProcess();
    public static native boolean deleteSystemApp(String pkgName);
    public static native boolean isAppRequiredBySystem(String pkgName);
    
    public void cleanCallback(Context ctx, int status, String data) {
        DeviceAPI.cleanCallback(ctx, status, data);
    }
}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/api/NativeAPI.kt
package com.rarnu.tools.neo.api

import android.content.Context
import android.util.Log

object NativeAPI {

    var jniLoaded = false
    init {
        try {
            System.loadLibrary("rarnucmd")
            jniLoaded = true
        } catch (t: Throwable) {
            Log.e("NativeAPI", "System.loadLibrary => $t")
        }
    }

    var rejected = false
    var systemRW = false

    external fun mount(): Boolean
    external fun isSystemRW(): Boolean
    external fun makePreferenceReadable(sdk: Int, packageName: String?)
    external fun freezeApplication(packageName: String?, isFreezed: Boolean): Boolean
    external fun freezeComponent(packageName: String?, componentName: String?, isFreezed: Boolean): Boolean
    external fun freezeComponents(packageName: String?, componentNames: Array<String>?, isFreezed: Boolean): Boolean
    external fun systemClean(ctx: Context?)
    external fun writeFile(ctx: Context?, filePath: String?, text: String?, perm: Int): Boolean
    external fun catFile(src: String?, dest: String?, perm: Int): Boolean
    external fun deleteFile(src: String?): Boolean
    external fun forceDeleteFile(path: String?)
    external fun forceDropCache()
    external fun killProcess()
    external fun deleteSystemApp(pkgName: String?): Boolean
    external fun isAppRequiredBySystem(pkgName: String?): Boolean

    external fun getBaseURL(): String?

    external fun freezeOnLoad()

    fun cleanCallback(ctx: Context?, status: Int, data: String?) {
        DeviceAPI.cleanCallback(ctx, status, data)
    }
}

